build.gradle

```diff
@@ -76,15 +74,14 @@ dependencies {
     api 'com.fasterxml.jackson.core:jackson-databind:2.11.0'
 
     // Preconditions
-    api 'com.google.guava:guava:29.0-jre'
+    api 'com.google.guava:guava:31.0-jre'
 
     // Testing libraries
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.0'
-    testCompileOnly 'junit:junit:4.13'
-    testRuntimeOnly 'org.junit.vintage:junit-vintage-engine'
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.1'
+    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.8.1'
```

Comment:
I agree the tests are a bit confusing, especially the dependencies.
Maybe, we can just use junit without `jupiter` and `vintage` but I guess we splitted the dependencies because of this: https://stackoverflow.com/questions/38822189/why-were-junit-jupiter-and-junit-vintage-separated-when-i-running-testcase-in-in

You removed `testRuntimeOnly 'org.junit.vintage:junit-vintage-engine'` but I wonder if it has any implication with the `includeEngines 'junit-vintage'` above in this gradle file.

@paulboocock do you have any memory of this?

Reply:
Nope, I think this was before my time 🤷